pkg/sac/test_scope_checker_core.go

```diff
@@ -0,0 +1,159 @@
+package sac
+
+import (
+	"context"
+	"testing"
+
+	"github.com/stackrox/rox/generated/storage"
+	"github.com/stackrox/rox/pkg/auth/permissions"
+	"github.com/stackrox/rox/pkg/sac/effectiveaccessscope"
+)
+
+// TestClusterScope contains cluster-level scope information (cluster included or list of included namespaces
+// in the cluster) to build test scope checkers.
+type TestClusterScope struct {
+	Namespaces []string
+	Included   bool
+}
+
+// TestResourceScope contains resource-level scope information (resource fully included or list of included clusters
+// and associated cluster-level scope information) to build test scope checkers.
+type TestResourceScope struct {
+	Clusters map[string]*TestClusterScope
+	Included bool
+}
+
+type testScopeCheckerCore struct {
+	scope map[storage.Access]map[permissions.Resource]*TestResourceScope
+	path  []ScopeKey
+}
+
+// TestScopeCheckerCoreFromAccessResourceMap creates a ScopeCheckerCore that allows full access to the input
+// (accessMode, Resource) pairs for testing purposes.
+func TestScopeCheckerCoreFromAccessResourceMap(_ *testing.T, targetResources []permissions.ResourceWithAccess) ScopeCheckerCore {
+	includedResourceScope := &TestResourceScope{
+		Included: true,
+	}
+	core := &testScopeCheckerCore{
+		scope: make(map[storage.Access]map[permissions.Resource]*TestResourceScope, 0),
+	}
+	for _, resource := range targetResources {
+		access := resource.Access
+		if _, accessExists := core.scope[access]; !accessExists {
+			core.scope[access] = make(map[permissions.Resource]*TestResourceScope, 0)
+		}
+		core.scope[access][resource.Resource.GetResource()] = includedResourceScope
+	}
+	return core
+}
+
+// TestScopeCheckerCoreFromFullScopeMap creates a ScopeCheckerCore that allows scoped access to the input
+// scope tree for testing purposes.
+func TestScopeCheckerCoreFromFullScopeMap(_ *testing.T, targetScope map[storage.Access]map[permissions.Resource]*TestResourceScope) ScopeCheckerCore {
```

Comment:
How about creating an alias for `map[storage.Access]map[permissions.Resource]*TestResourceScope` to make usage simpler? It's easier to type&read `sac.ScopeMap`